source/simulator/stokes_matrix_free.cc

```diff
@@ -1508,447 +1494,472 @@ namespace aspect
+                active_cell_data.viscosity(cell, 0)[i] = active_viscosity_vector(local_dof_indices[0]);
+              else
+                {
+                  fe_values_projection.reinit(DG_cell);
+                  fe_values_projection.get_function_values(active_viscosity_vector,
+                                                           local_dof_indices,
+                                                           values_on_quad);
+
+                  // Do not allow viscosity to be greater than or less than the limits
+                  // of the evaluated viscosity on the active level.
+                  for (unsigned int q=0; q<n_q_points; ++q)
+                    active_cell_data.viscosity(cell, q)[i]
+                      = std::min(std::max(values_on_quad[q], minimum_viscosity), maximum_viscosity);
+                }
+            }
+        }
+    }
+
+    active_cell_data.is_compressible = sim.material_model->is_compressible();
+    active_cell_data.pressure_scaling = sim.pressure_scaling;
+
+    // Store viscosity tables and other data into the active level matrix-free objects.
+    stokes_matrix.set_cell_data(active_cell_data);
+
+    if (sim.parameters.n_expensive_stokes_solver_steps > 0)
+      {
+        A_block_matrix.set_cell_data(active_cell_data);
+        Schur_complement_block_matrix.set_cell_data(active_cell_data);
+      }
+
+    const unsigned int n_levels = sim.triangulation.n_global_levels();
+    level_cell_data.resize(0,n_levels-1);
+
+    level_viscosity_vector = 0.;
+    level_viscosity_vector.resize(0,n_levels-1);
+
+    // Project the active level viscosity vector to multilevel vector representations
+    // using MG transfer objects. This transfer is based on the same linear operator used to
+    // transfer data inside a v-cycle.
+#if false
+    MGTransferMatrixFree<dim,GMGNumberType> transfer;
+    transfer.build(dof_handler_projection);
+
+    // Explicitly pick the version with template argument double to convert
+    // double-valued active_viscosity_vector to GMGNumberType-valued
+    // level_viscosity_vector:
+    transfer.template interpolate_to_mg<double>(dof_handler_projection,
+                                                level_viscosity_vector,
+                                                active_viscosity_vector);
+#else
+      const unsigned int min_level = 0;
+      const unsigned int max_level = n_levels-1;
+
+      MGLevelObject<
+      MGTwoLevelTransfer<dim, dealii::LinearAlgebra::distributed::Vector<GMGNumberType>>>
+      transfers(min_level, max_level);
+
+      for (unsigned int l = min_level; l < max_level; ++l)
+        transfers[l + 1].reinit(dofhandlers_projection[l + 1], dofhandlers_projection[l]);
+
+      MGTransferGlobalCoarsening<dim, dealii::LinearAlgebra::distributed::Vector<GMGNumberType>> transfer(transfers, [&](const auto l, auto &vec)
+      {
+        (void) l;
+        (void) vec;
+        Assert(false, ExcNotImplemented());
```

Comment:
Is there an operator we could use? Or should be setup the partitioners ourselves?